knowledge-base/adr/0020-generator-options.md

```diff
@@ -37,26 +37,35 @@ We should discuss the details of those defaults and make a concious decision on
+| :------------------------------------------------------- | :------------------------------- | :------------------------------------------------------------------------------------------------ | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `--inputDir`, `-i` _(required)_                          | `--input`, `-i` _(required)_     | Input directory/file                                                                              | _same_                                                                                                                                                                                                         |
+| `--outputDir`, `-o` _(required)_                         | `--outputDir`, `-o` _(required)_ | Output directory                                                                                  | _same_                                                                                                                                                                                                         |
+| `--clearOutputDir`                                       | `--clearOutputDir`               | Delete all files in output directory                                                              | _same_                                                                                                                                                                                                         |
+| `--generateJs`                                           | `--transpile`, `-t`              | Transpiles, default true                                                                          | Transpiles, default false. If set if tsconfig is enabled with default (unless configured). This should be explicitly stated in the documentation.                                                              |
+| `--generatePackageJson`                                  | `--[no]-packageJson`             | Writes a default package.json, default true.                                                      | Writes a default package.json, default true (needs core dependency). Optionally in the future: Writes a custom package.json if passed. Keep boolean for now.                                                   |
+| `--serviceMapping`                                       | `--perServiceConfig`             | Considers a service mapping file. Shows a warning if none is given (differs from OData behavior). | Only generates a per service configuration, if a file path is provided. No warning if it is not provided. If provided, `packageName` and `directoryName` will be generated, but are not required to be set. \* |
```

Comment:
```suggestion
| `--serviceMapping`                                       | `--perServiceConfig`             | Considers a per service configuration file. Shows a warning if none is given (differs from OData behavior). | Only generates a per service configuration, if a file path is provided. No warning if it is not provided. If provided, `packageName` and `directoryName` will be generated, but are not required to be set. \* |
```

Reply:
What about `--serviceConfig` as a shorter option?

Reply:
@FrankEssenberger What do you say?